helpers/helpers-spec_service.bal

```diff
@@ -0,0 +1,190 @@
+    }
+
+    # Returns all HTTP headers from the request
+    # + return - returns a map of headers or error
+    resource function get headers(http:Headers headers) returns specific_headers_response|Error_serverFailure {
+        do {
+            map<string> filteredHeaders = check getSpecificHeaders(headers = headers);
+            specific_headers_response response = {headers: filteredHeaders};
+            return response;
+        } on fail error e {
+            log:printError("Error retrieving headers", 'error = e, stackTrace = e.stackTrace());
+            Error_serverFailure response = {
+                body: {
+                    message: "Failed to retrieve headers",
+                    code: "err_007"
+                }
+            };
+            return response;
+        }
+    }
+
+    # Returns the value of the user-agent header
+    # The http:header annotation ensures that a default error message will be automatically
+    # created and returned if the header can't be found.
+    # + return - returns can be any of following types 
+    # http:Ok (Get a UUID V4.)
+    resource function get user\-agent(@http:Header string user\-agent) returns ua_response {
+        ua_response response = {user\-agent};
+        return response;
+    }
+
+    # Returns a unique ID as per UUID v4 spec
+    #
+    # + return - returns can be any of following types 
+    # http:Ok (Get a UUID V4.)
+    # http:Error_serverFailure (Response for any error)
+    resource function get uuid() returns uuid_response|Error_serverFailure {
+
+        do {
+            string tempUuid = uuid:createRandomUuid();
+            log:printDebug("Generated UUID: " + tempUuid);
+            uuid_response response = {"uuid": tempUuid};
+            return response;
+        } on fail error e {
+            log:printDebug("UUID Generated failed: " + e.toString());
+            Error_serverFailure response = {body: {"message": "failed to create UUID", "code": "err_001"}};
+            return response;
+        }
+    }
+
+    resource function post 'base64/decode/[string value]() returns Base64_responseOk|Error_responseBadRequest {
+        // Validate incoming string 
+        log:printDebug("Incoming text: " + value);
+
+        if (value.length() > MAX_BASE64_STRING_SIZE) {
+            Error_responseBadRequest response = {body: {"message": "String is too large. Sorry.", "code": "err_002"}};
+            return response;
+        }
+        string:RegExp pattern = re `^[0-9a-zA-Z=]+$`;
+
+        if ((value.matches(pattern)) is false) {
+            Error_responseBadRequest response = {body: {"message": "Invalid characters. Sorry.", "code": "err_003"}};
+            return response;
+        }
+        string|error decodedValue = mime:base64Decode(value).ensureType(string);
+
+        if (decodedValue is string) {
+            Base64_responseOk response = {body: {"value": decodedValue}};
+            log:printDebug("Decoded Value OK  " + decodedValue);
```

Comment:
The log functions also accept passing named arguments for additional context, can avoid concatenation that way - https://ballerina.io/learn/by-example/logging-with-context/